cranelift/codegen/src/isa/x64/lower.isle

```diff
@@ -1452,6 +1452,27 @@
 (rule (lower (trap code))
       (side_effect (x64_ud2 code)))
 
+;;;; Rules for `trapif` ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+;; The flags must not have been clobbered by any other instruction between the
+;; iadd_ifcout and this instruction, as verified by the CLIF validator; so we
+;; can simply use the flags here.
+(rule (lower (trapif cc flags @ (iadd_ifcout _ _) tc))
+      (let ((producer ProducesFlags (ProducesFlags.AlreadyExistingFlags flags))
+            (_ Unit (trap_if_bool (bool producer cc) tc)))
```

Comment:
here and elsewhere, we shouldn't do the side-effecting thing and invoke `trap_if_bool` just to throw the result away; we should return it. This can work similarly to other ops that don't return values most likely -- see e.g. stores, with the `(side_effect ...)` ctor.

Reply:
I'm not sure what value we would return here -- `TrapIf` and friends have no outputs. Are you suggesting that `trap_if_bool` should return an `InstOutput` and always return `(output_none)`?

Reply:
Ah of course: we can return a `SideEffectNoResult.Inst2` instead. I'll make that change, it's much cleaner than relying on the effects currently.